### DISCUSSION TITLE
Does this library support usage of a custom language's server worker?

### DISCUSSION THREAD

**Discussion by Lemour-sudo** [suren-atoyan/monaco-react#594] [General]
I see that we can use `getWorkerUrl` to run a custom worker.

However, I am not sure how to link this to the `Editor` component as the `language` property in the component is said to be for supported certain languages: https://github.com/microsoft/monaco-editor/tree/main/src/basic-languages
Is it possible to just have a custom language's server worker integrated with this library's editor instead of relying on the hardcoded keywords and tokens?

Moreover, the `label` argument being passed to `getWorker` does not detect a custom registered language id. In the code snippet below, the logged label is just "editorWorkerService" instead of the registered "lingo" language id. Is this how it is meant to be used?

This is the snippet I have running using Vite:

```
import { useEffect } from "react";

import { loader } from "@monaco-editor/react";
import Editor, { useMonaco } from "@monaco-editor/react";
import editorWorker from "monaco-editor/esm/vs/editor/editor.worker?worker";

const languageId = "lingo";

export function App() {
  const monaco = useMonaco();

  useEffect(() => {
    if (monaco) {
      monaco.languages.register({ id: languageId });

      self.MonacoEnvironment = {
        getWorker(_, label) {
          console.log("label:", label);
          return new editorWorker();
        },
      };

      loader.config({ monaco });
    }
  }, [monaco]);

  return (
    <Editor
      height="90vh"
      defaultValue="// some comment"
      language={languageId}
    />
  );
}

```

**Comment by suren-atoyan** [comment]
related #622

**Comment by Lemour-sudo** [comment]
Yes, I decided to create an issue from this. I believe there might be someone who has recently tried the approach mentioned here.

**Reply by sachinaryal2077** [reply]
> Yes, I decided to create an issue from this. I believe there might be someone who has recently tried the approach mentioned here.

Did you find a solution ?
